src/passes/Inlining.cpp

```diff
@@ -666,8 +674,17 @@ struct FunctionSplitter {
         assert(iff->ifTrue->type == Type::unreachable);
       }
     }
-
     // Success, this matches the pattern.
+
+    // If outlined function will be worth normal inline, skip the intermediate
+    // state and inline fully.
+    if (numIfs == 1) {
+      Index outlinedFunctionSize = Measurer::measure(iff->ifTrue);
```

Comment:
I don't follow this. We outline the body of the if, not the condition?

Reply:
My understanding is with single if condition PatternB looks like:

```
  if (<condition>) {
    outlined(...);
  }
```

Did I get it wrong?

Reply:
Ah, sorry, I read that wrong. You are correct. I read it as measuring `iff->condition` but your code is right as it has `iff->ifTrue`.